aiida_gaussian/tests/parsers/test_base.py

```diff
@@ -0,0 +1,57 @@
+"""Tests for the :class:`aiida_gaussian.parsers.gaussian.GaussianBaseParser` class."""
+# pylint: disable=redefined-outer-name
+import pytest
+from aiida.orm import Dict
+
+
+def recursive_array_to_list(data):
+    import numpy as np
+
+    if isinstance(data, dict):
+        return {key: recursive_array_to_list(value) for key, value in data.items()}
+    elif isinstance(data, list):
+        return [recursive_array_to_list(value) for value in data]
+    elif isinstance(data, np.ndarray):
+        return data.tolist()
+    else:
+        return data
+
+
+@pytest.fixture
+def inputs():
+    return {
+        "parameters": Dict(
+            {
+                "link0_parameters": {
+                    "%chk": "aiida.chk",
+                    "%nprocshared": "1",
+                },
+                "functional": "BLYP",
+                "basis_set": "6-31g",
+                "charge": 0,
+                "multiplicity": 1,
+                "route_parameters": {
+                    "scf": {"maxcycle": 512, "cdiis": None},
+                    "nosymm": None,
+                    "opt": None,
+                },
+            }
+        )
+    }
+
+
+def test_nan_inf(generate_calc_job_node, generate_parser, inputs, data_regression):
+    """Test parsing a case where the parsed dictionary contains ``nan`` and ``inf`` values."""
+    node = generate_calc_job_node("gaussian", "base", "nan_inf", inputs=inputs)
```

Comment:
Here, if I understand correctly, the `inputs` are not used and they don't match with the output of the test (e.g. functional `BLYP` vs `M062X`). This is a bit confusing to me. Isn't it better to leave `inputs=None`?

Reply:
Yes, you are right. I am not sure anymore why I added this. Sometimes the parser plugins check the inputs of a node, so in that case you should set them in the test or it will fail. But here this is not the case. I have removed it and the test passes just fine.